### PR TITLE
Preconnections are just structs

### LINKED ISSUE #859: When are objects frozen?
Discussion at the 11 Jun 2021 Interim:

The content of any objects passed as part of NewPreconnection(), Initiate(), Listen(), or Rendezvous() is frozen at the moment of Initiate/Listen/Rendezvous.

Thus, objects sent to NewPreconnection (Endpoints and Properties) are mutable, and it might be possible (depending on the implementation) to change them in the preconnection by modifying the object directly. Applications should either
(1) Instantiate the connection before modifying Endpoint or Property objects for another preconnection.
(2) If there are open preconnections that will initiate further connections, create new instances of Properties and Endpoints to modify their attributes in a different preconnection.

There are other ways to do this. For example, NewPreconnection() could do a deep copy of the properties and endpoints, and therefore reusing or freeing the argument objects would always be safe. But this was not the sense of the group at the interim.

While this issue is uncomfortably close to the "implementation dependent", not having fixed assumptions about this increasing the friction of switching platforms using TAPS, and would probably be the source of nasty bugs.

### PR DESCRIPTION
**Pull request by mwelzl**
Closes #859

This is quite minimal... meant as a start. Or is this already enough?

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -436,6 +436,14 @@ Connection.Close()
 Connection2.Close()
 ~~~
 
+Preconnections are reusable. Hence, for example, after the Connections were closed,
+the following would be correct:
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:440]: After Initiated wouldn't this also be OK?

Comment (mwelzl) [reply draft-ietf-taps-interface.md:440]: Of course - anywhere, really, it's just a struct. But this gets a bit harder to write about ("a similar line would also work below the 'initiate' call higher above"), unless we use more code just to show this; I think, with the explanation below, it should be clear enough as it is.

**Review by tfpauly — APPROVED** [review]
I'm good with this!

File: draft-ietf-taps-interface.md (hunk 1 above)
Comment (gorryfair) [reply draft-ietf-taps-interface.md:440]: Could we try: 
"Preconnections are reusable after being used to initiate a Connection. Hence, for example, after the Connections were closed,
 the following would be correct:"

Comment (mwelzl) [reply draft-ietf-taps-interface.md:440]: Your wish has been granted.

**Review by gorryfair — APPROVED** [review]
Looks good to me...

**Pull request merged**